src/einsteinpy/symbolic/weyl.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import sympy
+
+from einsteinpy.symbolic.ricci import RicciScalar, RicciTensor
+from einsteinpy.symbolic.riemann import RiemannCurvatureTensor
+from einsteinpy.symbolic.tensor import Tensor
+
+
+class WeylTensor(Tensor):
+    """
+    Class for defining Weyl Tensor
+    """
+
+    def __init__(self, arr, syms, config="llll", parent_metric=None):
+        """
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+            Sympy Array or multi-dimensional list containing Sympy Expressions
+        syms : tuple or list
+            Tuple of crucial symbols dentoting time-axis, 1st, 2nd, and 3rd axis (t,x1,x2,x3)
+        config : str
+            Configuration of contravariant and covariant indices in tensor. 'u' for upper and 'l' for lower indices. Defaults to 'llll'.
+        parent_metric : ~einsteinpy.symbolic.metric.WeylTensor
+            Corresponding Metric for the Weyl Tensor. Defaults to None.
+
+        Raises
+        ------
+        TypeError
+            Raised when arr is not a list or sympy Array
+        TypeError
+            syms is not a list or tuple
+        ValueError
+            config has more or less than 3 indices
+
+        """
+        super(WeylTensor, self).__init__(arr, config=config)
+        self.syms = syms
+        self._order = 4
+        self._parent_metric = parent_metric
+        if isinstance(syms, (list, tuple)):
+            self.syms = syms
+            self.dims = len(self.syms)
+        else:
+            raise TypeError("syms should be a list or tuple")
+        if not len(config) == self._order:
+            raise ValueError("config should be of length {}".format(self._order))
+
+    @classmethod
+    def from_metric(cls, metric, dims, parent_metric=None):
+        """
```

Comment:
```
from_metric(cls, metric)
```

No need for other arguments. There is no point of having `parent_metric` when the quantity is initialized from a metric itself. No need for `dims` also, as it can be obtained from `metric.dims`